portfolio/src/main/webapp/script.js

```diff
@@ -29,18 +32,34 @@ function addRandomFact() {
   factContainer.innerText = fact;
 }
 
+/**
+ * Fetches comments and displays them to the page.
+ */
 function getComments() {
   document.getElementById('comments-container').innerHTML = '';
-  fetch('/data').then(response => response.json()).then((tasks) => {
-    const taskListElement = document.getElementById('comments-container');
-    tasks.forEach((task) => {
-      taskListElement.appendChild(createListElement(task.comment));
-    })
+  let num = document.getElementById('num');
+  num = num.options[num.selectedIndex].value;
+  const url = '/data?limit=' + num;
+
+  fetch(url).then((response) => response.json()).then((comments) => {
+    const commentsListElement = document.getElementById('comments-container');
+    comments.forEach((comment) => {
+      commentsListElement.appendChild(
+          createListElement(comment.name, comment.comment));
+    });
   });
 }
 
-function createListElement(comment) {
+/**
+ * Deletes all comments from the page.
+ */
+function deleteComments() {
+  const request = new Request('/delete-data', {method: 'POST'});
+  fetch(request).then((result) => getComments());
+}
+
+function createListElement(name, comment) {
   const liElement = document.createElement('p');
-  liElement.innerText = comment;
+  liElement.innerText = name + ': ' + comment;
```

Comment:
do you want to also include the timestamp here? you could also format it "X seconds ago"

Reply:
done. formatted postTime with date and time each comment was posted.